contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -2883,7 +2883,23 @@ static void bbf_ProcessUtility(PlannedStmt *pstmt,
 					if (HeapTupleIsValid(tuple))
 						roleform = (Form_pg_authid) GETSTRUCT(tuple);
 					else
-						continue;
+					{
+						/* Supplied login name might be in windows format i.e, domain\user form */
+						tuple = get_roleform_ext(role_name);
```

Comment:
You need to release the syscache once you're done.

Reply:
Also, it's not a good practice to return a tuple which needs to be released outside the function that pinned it. Return the role_name instead. And add this here,
```
role_upn_name = get_upnform_ext(role_name);
if (role_upn_name != NULL)
{
   pfree(rolename);
   rolspec->rolename = role_upn_name;
}
else
    continue;
```